tests/rptest/tests/redpanda_oauth_test.py

```diff
@@ -184,6 +187,70 @@ def check_sr_subjects():
 
         wait_until(check_sr_subjects, timeout_sec=5)
 
+    @cluster(num_nodes=4)
+    def test_java_client(self):
+        kc_node = self.keycloak.nodes[0]
+
+        self.keycloak.admin.create_user('norma',
+                                        'desmond',
+                                        realm_admin=True,
+                                        email='10086@sunset.blvd')
+        self.keycloak.login_admin_user(kc_node, 'norma', 'desmond')
+        self.keycloak.admin.create_client(CLIENT_ID)
+
+        # add an email address to myapp client's service user. this should
+        # appear alongside the access token.
+        self.keycloak.admin.update_user(f'service-account-{CLIENT_ID}',
+                                        email='myapp@customer.com')
+
+        service_user_id = self.keycloak.admin_ll.get_user_id(
+            f'service-account-{CLIENT_ID}')
+
+        self.rpk.create_topic(EXAMPLE_TOPIC)
+        expected_topics = set([EXAMPLE_TOPIC])
+        wait_until(lambda: set(self.rpk.list_topics()) == expected_topics,
+                   timeout_sec=5)
+
+        cfg = self.keycloak.generate_oauth_config(kc_node, CLIENT_ID)
+        cli = KafkaCliTools(self.redpanda, oauth_cfg=cfg)
+
+        self.redpanda.logger.debug(
+            "Without an appropriate ACL, topic list empty, produce fails with an authZ error."
+        )
+
+        with expect_exception(AuthorizationError, lambda _: True):
+            cli.oauth_produce(EXAMPLE_TOPIC, 1)
+
+        assert len(cli.list_topics()) == 0
+
+        self.redpanda.logger.debug(
+            "Grant access to service user. We can see it in the list and produce now."
+        )
+
+        self.rpk.sasl_allow_principal(f'User:{service_user_id}', ['all'],
+                                      'topic', EXAMPLE_TOPIC, self.su_username,
+                                      self.su_password, self.su_algorithm)
+
+        assert set(cli.list_topics()) == set(expected_topics)
+
+        N_REC = 10
+        cli.oauth_produce(EXAMPLE_TOPIC, N_REC)
+        records = []
+        for i in range(0, N_REC):
+            rec = self.rpk.consume(EXAMPLE_TOPIC, n=1, offset=i)
+            records.append(json.loads(rec))
```

Comment:
Is the output of consuming multiple records a bit tricky to process?

Reply:
Yeah exactly. At one point I had some weird thing to munge it into a json array, but it's beside the point of this test.